example/lib/main.dart

```diff
@@ -245,57 +249,81 @@ class _MyHomePageState extends State<MyHomePage> {
-            ),
-            "tr": Style(
-              border: Border(bottom: BorderSide(color: Colors.grey)),
-            ),
-            "th": Style(
-              padding: EdgeInsets.all(6),
-              backgroundColor: Colors.grey,
+        child: Column(
+          children: [
+            Html(
+              data: htmlData,
+              style: {
+                "table": Style(
+                  backgroundColor: Color.fromARGB(0x50, 0xee, 0xee, 0xee),
+                ),
+                "tr": Style(
+                  border: Border(bottom: BorderSide(color: Colors.grey)),
+                ),
+                "th": Style(
+                  padding: EdgeInsets.all(6),
+                  backgroundColor: Colors.grey,
+                ),
+                "td": Style(
+                  padding: EdgeInsets.all(6),
+                  alignment: Alignment.topLeft,
+                ),
+              },
+              customRender: {
+                "table": (context, child) {
+                  return SingleChildScrollView(
+                    scrollDirection: Axis.horizontal,
+                    child:
+                        (context.tree as TableLayoutElement).toWidget(context),
+                  );
+                }
+              },
+              customImageRenders: {
+                networkSourceMatcher(domains: ["flutter.dev"]):
+                    (context, attributes, element) {
+                  return FlutterLogo(size: 36);
+                },
+                networkSourceMatcher(domains: ["mydomain.com"]):
+                    networkImageRender(
+                  headers: {"Custom-Header": "some-value"},
+                  altWidget: (alt) => Text(alt ?? ""),
+                  loadingWidget: () => Text("Loading..."),
+                ),
+                // On relative paths starting with /wiki, prefix with a base url
+                (attr, _) =>
+                        attr["src"] != null && attr["src"]!.startsWith("/wiki"):
+                    networkImageRender(
+                        mapUrl: (url) => "https://upload.wikimedia.org" + url!),
+                // Custom placeholder image for broken links
+                networkSourceMatcher():
+                    networkImageRender(altWidget: (_) => FlutterLogo()),
+              },
+              onLinkTap: (url, _, __, ___) {
+                print("Opening $url...");
+              },
+              onImageTap: (src, _, __, ___) {
+                print(src);
+              },
+              onImageError: (exception, stackTrace) {
+                print(exception);
+              },
             ),
-            "td": Style(
-              padding: EdgeInsets.all(6),
-              alignment: Alignment.topLeft,
+            Html(
+              data: htmlDataLines,
+              style: {
+                'html': Style(maxLine: 1, textOverflow: TextOverflow.ellipsis),
+              },
```

Comment:
I see what you're trying to do here but I would suggest just adding the maxLines HTML to the original widget. Then in `Style()` you can do `'h3': Style(maxLines: 2, textOverflow: TextOverflow.ellipsis)` to show off that feature, without needing multiple HTML widgets on the page.

Reply:
Oh yes the example code here isn't the best to show of this feature. But I like the implementation itself. @nguyenxdat could you follow @tneotia 's suggestion and update the example app?